src/lib/core/datetime/native-date-adapter.ts

```diff
@@ -0,0 +1,141 @@
+import {DateAdapter} from './date-adapter';
+
+
+// TODO(mmalerba): Remove when we no longer support safari 9.
+/** Whether the browser supports the Intl API. */
+const SUPPORTS_INTL_API = typeof Intl != 'undefined';
+
+
+/** Creates an array and fills it with values. */
+function range<T>(length: number, valueFunction: (index: number) => T): T[] {
+  return Array.apply(null, Array(length)).map((v: undefined, i: number) => valueFunction(i));
+}
+
+
+/** Adapts the native JS Date for use with cdk-based components that work with dates. */
+export class NativeDateAdapter extends DateAdapter<Date> {
+  getYear(date: Date): number {
+    return date.getFullYear();
+  }
+
+  getMonth(date: Date): number {
+    return date.getMonth();
+  }
+
+  getDate(date: Date): number {
+    return date.getDate();
+  }
+
+  getDayOfWeek(date: Date): number {
+    return date.getDay();
+  }
+
+  getMonthNames(style: 'long' | 'short' | 'narrow'): string[] {
+    if (SUPPORTS_INTL_API) {
+      let dtf = new Intl.DateTimeFormat(this.locale, {month: style});
+      return range(12, i => dtf.format(new Date(2017, i, 1)));
+    }
+    return {
+      'long': [
+        'January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September',
+        'October', 'November', 'December'
+      ],
+      'short': ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'],
+      'narrow': ['J', 'F', 'M', 'A', 'M', 'J', 'J', 'A', 'S', 'O', 'N', 'D']
```

Comment:
By hard-coding the values into the return, you'll end up creating a new instance of these arrays each time the method is called. Move this object into a constant?